back/engines/commercial/nlp/app/services/nlp/api.rb

```diff
@@ -8,17 +8,19 @@ class Api
 
     LONG_TIMEOUT = 2 * 60 # 2 minutes
 
+    attr_reader :authorization_token
     delegate :post, :base_uri, :get, to: :class
 
-    def initialize(base_uri = ENV.fetch('CL2_NLP_HOST'))
-      base_uri(base_uri)
+    def initialize(base_uri: nil, authorization_token: nil)
+      @authorization_token = authorization_token || ENV.fetch('NLP_API_TOKEN')
+      base_uri(base_uri || ENV.fetch('CL2_NLP_HOST'))
     end
 
     def update_tenant(dump)
       post(
         '/v1/tenants',
         body: dump.to_json,
-        headers: { 'Content-Type' => 'application/json' },
+        headers: authorization_header.merge('Content-Type' => 'application/json'),
```

Comment:
I don't know the NLP API very well (yet! :D), do we have endpoints that do not require the `Content-Type` header set to json?

If (almost) all endpoints use them, we could think about adding a private method that handles the merge for us:

```ruby

# [...]

private

def headers
  authorization_header.merge('Content-Type' => 'application/json')
end


```

Reply:
Yep, I also hesitated there, but there is indeed some endpoints (GET's) that do not require the `Content-Type` header. I guess it would not hurt, but the NLP API can be a bit skittish. I'll test it out if I find some time.